Source/Applications/Web/NavigationStructure.tsx

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Navigation, NavigationActionBar, NavigationButton, NavigationGroup } from '@shared/portal';
+import React from 'react';
+import { container } from 'tsyringe';
+import { ApplicationModel } from './ApplicationModel';
+
+export type NavigationStructureProps = {
+    applications?: ApplicationModel[];
+    handleNavbarActionButtonClick?: () => void;
+};
+
+export function NavigationStructure(props: NavigationStructureProps) {
+    const navigation = container.resolve(Navigation);
+    const navigationItems = props.applications?.map(
+        (app) => ({ name: app.name, items: [{ name: 'Default' }] } as NavigationGroup)
+    );
+
+    // Will this create a new callback on every state change?
+    // Does it matter?
+    const navigationActionBar =  {
+        button: new NavigationButton('New Application', 'Add', () =>
+            props?.handleNavbarActionButtonClick?.()
+        ),
+        placement: 'bottom',
+    } as NavigationActionBar;
+
+    navigation.set(navigationItems ?? [], navigationActionBar);
```

Comment:
Based on "rate of change", it may make more sense to keep the NavigationActionBar as a completely separate component from the NavigationGroups. Both reside in the Navigation area, but both may need to change at different cadences.

Thoughts?